protocol/contracts/farm/facets/MarketplaceFacet/Dynamic.sol

```diff
@@ -0,0 +1,241 @@
+/**
+ * SPDX-License-Identifier: MIT
+ **/
+
+ pragma solidity =0.7.6;
+ pragma experimental ABIEncoderV2;
+ 
+ import "../../../libraries/Token/LibTransfer.sol";
+ import "./PodTransfer.sol";
+ import "hardhat/console.sol";
+ 
+ 
+ /* 
+ * @author: Malteasy
+ * @title: Dynamic Pricing
+ */
+ 
+ contract Dynamic is PodTransfer { 
+
+    using SafeMath for uint256;
+
+    enum EvaluationMode {
+        Fixed,
+        Dynamic
+    }
+    /**
+        The polynomial's constant terms are split into: 1) constant * 10^base , 2) the base the constant is raised to and, 3) the sign of the constants.
+        Example conversion to PiecewisePolynomial struct: 
+
+            Range(0, 1) -> Polynomial(0.25*x^3 + 25*x^2 + x + 1)
+            Range(1, 2) -> Polynomial(0.0125*x^3 + 50*x^2 + x - 2)
+            Range(2, Infinity) -> Polynomial(-1)
+            
+        Resulting PiecewisePolynomial:
+
+            breakpoints: [0, 1, 2, 0, 0, ... , 0]
+            constants: [1, 1, 25, 25, 2, 1, 50, 125, 1, 0, 0, ... , 0]
+            (expanded) bases: [0, 0, 0, 2, 0, 0, 0, 4, 0, 0, 0, ... , 0]
+            (expanded) signs: [true, true, true, true, false, true, true, true, false, false, false, ... , false]
+        
+    */
+    
+    //
+    struct PiecewisePolynomial {
+        uint256[16] breakpoints; //The breakpoints for the piecewise. The last breakpoint in the array is the final interval for the polynomial, extending to infinity.
+        uint256[64] constants; // [c0_0,c1_0,c2_0,c3_0, ... , c0_31,c1_31,c2_31,c3_31] Contains the coefficients of the polynomial, concatenated according to their piecewise index. The first coefficient is the constant term, the second is the first term, etc.  
+        uint256[2] packedBases; // [b0,b1,b2,b3] b0 contains the 8-bit bases for polynomials 0-7, b1 for polynomials 8-15
```

Comment:
these are not actually the bases, this is the exponent part of the number

Reply:
base is 10

Reply:
![image](https://user-images.githubusercontent.com/94831127/186542313-5b2df808-8d25-4028-8790-da19459cb53f.png)